### PR TITLE
Clipboard

### PR DESCRIPTION
**Pull request by jpmedley**
I don't have a link to back this up. It came from a conversation with the spec author and Chrome engineer who implemented it.

### CODE REVIEW

File: api/Clipboard.json [hunk 1]
```diff
@@ -74,7 +80,9 @@
           "support": {
             "chrome": [
               {
-                "version_added": "76"
+                "version_added": "76",
```
Comment (ddbeck) [inline api/Clipboard.json:83]: Is this really a partial implementation? Reading the spec, it's not clear to me that implementations are obligated to support all MIME types. Maybe you have more familiarity with the spec. If it is not obligatory, then I'd guess we ought to redo this as a single entry, without partial and the notes are formatted as:

1. `"From version 76, the <code>image/png</code> MIME type is supported."`
2. `"Before version 76, images are not supported."`

If implementations are supposed to be supporting arbitrary MIME types (or some set of MIME types,) then I think the existing structure is fine, though I might revise the note for the current version to stand alone a little better, perhaps `"Only <code>image/png</code> MIME type images are supported."`

Comment (jpmedley) [reply api/Clipboard.json:83]: For `write()`, this turns out to be browser specific. For `read()` it's not.

>These data types must be exposed by paste events if a corresponding native type exists on the clipboard. ([Link](https://w3c.github.io/clipboard-apis/#reading-from-clipboard))

I've changed the way I did `write()`, but not `read()`.

File: api/Clipboard.json [hunk 2]
```diff
@@ -6,7 +6,9 @@
         "support": {
           "chrome": [
             {
-              "version_added": "76"
+              "version_added": "76",
```
[the remark this answers is not in the document]
Comment (jpmedley) [reply api/Clipboard.json:9]: Done.

File: api/Clipboard.json [hunk 3]
```diff
@@ -219,30 +192,14 @@
...
-              },
-              {
-                "version_added": "66",
-                "partial_implementation": true,
-                "notes": "Images are not supported."
-              }
-            ],
+            "chrome": {
+              "version_added": "66",
+              "notes": "Version 76 added support for the <code>image/png</code> MIME type."
```
Comment (ddbeck) [inline api/Clipboard.json:197]:
```suggestion
              "notes": "From version 76, the <code>image/png</code> MIME type is supported."
```

File: api/Clipboard.json [hunk 4]
```diff
@@ -74,7 +53,9 @@
           "support": {
             "chrome": [
               {
-                "version_added": "76"
+                "version_added": "76",
+                "partial_implementation": true,
+                "notes": "Adds support for the <code>image/png</code> MIME type."
```
Comment (ddbeck) [inline api/Clipboard.json:58]:
```suggestion
                "notes": "From version 76, the <code>image/png</code> MIME type is supported."
```

File: api/Clipboard.json [hunk 5]
```diff
@@ -84,7 +65,9 @@
             ],
             "chrome_android": [
               {
-                "version_added": "84"
+                "version_added": "84",
+                "partial_implementation": true,
+                "notes": "Adds support for the <code>image/png</code> MIME type."
```
Comment (ddbeck) [inline api/Clipboard.json:70]:
```suggestion
                "notes": "From version 84, the <code>image/png</code> MIME type is supported."
```

File: api/Clipboard.json [hunk 6]
```diff
@@ -207,26 +192,14 @@
...
-                "notes": "Images are not supported."
-              }
-            ],
+            "chrome": {
+              "version_added": "66",
+              "notes": "Version 76 added support for the <code>image/png</code> MIME type."
+            },
+            "chrome_android": {
+              "version_added": "66",
+              "notes": "Version 84 added support for the <code>image/png</code> MIME type."
```
Comment (ddbeck) [inline api/Clipboard.json:201]:
```suggestion
              "notes": "From version 84, the <code>image/png</code> MIME type is supported."
```

File: api/Clipboard.json [hunk 7]
```diff
@@ -270,16 +243,10 @@
...
-              },
-              {
-                "version_added": "66",
-                "partial_implementation": true,
-                "notes": "Images are not supported."
-              }
-            ]
+            "webview_android": {
+              "version_added": "66",
+              "notes": "Version 84 added support for the <code>image/png</code> MIME type."
```
Comment (ddbeck) [inline api/Clipboard.json:248]:
```suggestion
              "notes": "From version 84, the <code>image/png</code> MIME type is supported."
```

File: api/Clipboard.json [hunk 8]
```diff
@@ -137,7 +120,9 @@
             },
             "webview_android": [
               {
-                "version_added": "84"
+                "version_added": "84",
+                "partial_implementation": true,
+                "notes": "Adds support for the <code>image/png</code> MIME type."
```
Comment (ddbeck) [inline api/Clipboard.json:125]:
```suggestion
                "notes": "From version 84, the <code>image/png</code> MIME type is supported."
```

**Comment by jpmedley** [comment]
Updated.

**Pull request merged**